Libraries/Components/WebView/WebView.ios.js

```diff
@@ -462,6 +462,38 @@ var WebView = React.createClass({
   },
 
   /**
+   * Evaluate JavaScript on the current page.
+   */
+  evaluateJavaScript: function(
+    script: string,
+    callback?: ?(error: ?Error, result: ?string) => void,
+  ): Promise {
+    return new Promise((resolve, reject) => {
+      var escaped = JSON.stringify(script).replace(/\u2028/g, '\\u2028').replace(/\u2029/g, '\\u2029');
+      var wrapped = 'JSON.stringify(eval(' + escaped + '))';
+
+      RCTWebViewManager.evaluateJavaScript(this.getWebViewHandle(), wrapped, function(error, result) {
+        if (!error) {
+          if (result === '') {
+            error = new Error("Error evaluating script.")
```

Comment:
@thebnich Ah, ok. Anyways, I still think we should move this to native side. Creating the error on the native side allows us to specify an error code also, which is nice. And we can get rid of the promise boilerplate if we use a promise instead of a callback in native side.